### PR TITLE
[Enhancement]: use enum for defining valid actions

### LINKED ISSUE #7: [Enhancement]: Define valid action enums for `manual mode`.
### Verified issue does not already exist?

- [X] I have searched and found no existing issue

### Suggested new feature or improvement

Currently the valid action is hardcoded into the codebase.

```ts
// src/modes/manual.mode/manual.ts
   
async recordEntry(sessionInfo: Session, data: any): Promise<Session> {
    ...
     if (validatedRecord.action === 'mark') {
            ...
     }
}
```

These values should be defined in an `_enum.ts` file. It shall define the mode specific actions.

### Scenario

This would help to improve the maintainability of the codebase.

### PR DESCRIPTION
**Pull request by ishan21sh**
### New

-   Added Action enum in manual mode

## Detailed description of the pull request / Additional comments
This adds _enum.ts for manual mode. The enum is used to define the valid actions and use it at the places where hardcoded string were being used earlier. I think this should close the mentioned issue. Looking forward to getting some reviews on this one. 
Thanks in advance.

## PR checklist
- [x] Closes #7
- [x] Feature or Enhancement
- [ ] Documentation updated

### CODE REVIEW

File: src/modes/manual.mode/_enum.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+enum Action {
+    MARK = "mark",
+}
+
+export { Action };
```
Comment (th3c0d3br34ker) [inline src/modes/manual.mode/_enum.ts]: Please make it as a `default` export.

**Review by th3c0d3br34ker — CHANGES_REQUESTED** [review]
Please resolve the added comments.

**Comment by th3c0d3br34ker** [comment]
thanks @ishan21sh on your submission. common feedback -- please comment first on the issue that you wish to work on, before you are starting on it. this helps any other contributor understand that you have picked up the issue.

**Comment by ishan21sh** [comment]
Yeah I will make the required changes thank you for the feedback and sorry for the delay was busy with some other assignments

**Comment by th3c0d3br34ker** [comment]
hey @ishan21sh please take note --
1. please name the variable as `ACTIONS` to follow a standard coding style.
2. update the usage of the `Action` in [manual.ts](https://github.com/kalviumcommunity/whosin/pull/8/files#diff-9bedced141656d2bb80befb5be8c59f143c2fee09c4c0ad95ab70d6976a947a8) as well.
3. please run `npm run format` before committing your code to for formating.

**Review by th3c0d3br34ker — CHANGES_REQUESTED** [review]
Please resolve the added comments.

**Comment by ishan21sh** [comment]
Made the required Improvements
1. Changed he variable name 'ACTION' to 'ACTIONS'
2. Added formatting using npm run format

**Review by th3c0d3br34ker — CHANGES_REQUESTED** [review]
hey @ishan21sh, suggestion -- 
1. the variable `ACTIONS` is expected to be capitalized.
2. please don't commit changes to files where you haven't made any changes like `package-lock.json` and `requestLogger.ts`

File: package-lock.json
Comment (th3c0d3br34ker) [inline package-lock.json:1]: please reset this file.

File: src/middlewares/requestLogger.ts
Comment (th3c0d3br34ker) [inline src/middlewares/requestLogger.ts:1]: please reset this file.

**Review by th3c0d3br34ker — CHANGES_REQUESTED** [review]

**Review by th3c0d3br34ker — APPROVED** [review]
thanks @ishan21sh for your support. this PR looks good to be merged. ✨

**Comment by ishan21sh** [comment]
Thank you sir for the feedbacks and opportunity

**Pull request merged**